sos/report/plugins/snap.py

```diff
@@ -28,4 +29,15 @@ def setup(self):
         self.add_service_status("snapd")
         self.add_journal(units="snapd")
 
+    def postproc(self):
+        self.do_file_sub(
+            "/var/lib/snapd/state.json",
+            (r"\"(macaroon|store-macaroon|key-id|session-macaroon"
```

Comment:
Should this feature have a test? Also I would feel slightly better if this would use a json parser and not just a regexp?

Reply:
All plugin file postprocessing is done via regex - we don't use file specific parsers. For one, what if the underlying issue that the sos report is being generated for is malformed json? The parser would fail during post processing and we'd likely end up with sensitive data being shipped.